src/ctl/ctl.c

```diff
@@ -78,16 +80,34 @@ char *Strdup(const char *s) {
     return p;
 }
 
+int umf_ctl_get(const char *name, void *ctx, void *arg) {
+    return ctl_query(NULL, ctx, CTL_QUERY_PROGRAMMATIC, name, CTL_QUERY_READ,
```

Comment:
if this is the implementation of the public API you should check args here and return errors

Reply:
Done

Reply:
I was thinking more of checking args and return the "INVALID_ARGUMENT" error. Btw there should be test for that (could be done in next PR as an effort to increase codecov)

Reply:
I changed asserts into returning `UMF_RESULT_ERROR_INVALID_ARGUMENT`. I can add test for those cases in separate PR.